src/main/java/uk/sky/cirrus/locking/Lock.java

```diff
@@ -0,0 +1,174 @@
+package uk.sky.cirrus.locking;
+
+import com.datastax.driver.core.*;
+import com.datastax.driver.core.exceptions.DriverException;
+import com.datastax.driver.core.exceptions.WriteTimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.sky.cirrus.locking.exception.CannotAcquireLockException;
+import uk.sky.cirrus.locking.exception.CannotReleaseLockException;
+
+import java.time.Duration;
+import java.util.UUID;
+
+/**
+ * Class to acquire a lock on cassandra table for a single application instance.  Each instance is given a unique identifier
+ * and attempts to gain a lock on the table.  If no lock is currently acquired, the instance is given the lock on the table, otherwise
+ * the instance must wait until the lock has been relinquished. If a lock cannot be acquired within the configured timeout
+ * interval, an exception is thrown.
+ */
+public class Lock {
+
+    private static final Logger log = LoggerFactory.getLogger(Lock.class);
+
+    private final String name;
+    private final Session session;
+    private final UUID client;
+    private boolean released;
+
+    private Lock(String name, Session session, UUID client) {
+        this.name = name;
+        this.session = session;
+        this.client = client;
+    }
+
+    /**
+     * @param clientId   Identifier for the owner of the lock
+     * @param lockConfig {@code LockConfig} for configuring the lock to migrate the schema
+     * @param keyspace   Name of the keyspace which is to be used for migration
+     * @param session    Cassandra {@code Session} to use while acquiring the lock
+     * @return the {@code Lock} object
+     * @throws CannotAcquireLockException if instance cannot acquire lock within the specified time interval or execution of query to insert lock fails
+     */
+    static Lock acquire(LockConfig lockConfig, String keyspace, Session session, UUID clientId) {
+
+        final String clientName = keyspace + ".schema_migration";
+        int acquireAttempts = 1;
+
+        log.info("Attempting to acquire lock for '{}', using client id '{}'", clientName, clientId);
+        long startTime = System.currentTimeMillis();
+
+        final Statement query = new SimpleStatement("INSERT INTO locks.locks (name, client) VALUES (?, ?) IF NOT EXISTS", clientName, clientId);
```

Comment:
That statement is executed in a loop. Also once this method is not static, the Lock instance could be reused.